src/libs/workspace-utils.ts

```diff
@@ -158,3 +158,11 @@ export const isValidWsExportTarget = safeCurry((sourceWs: WorkspaceWrapper, dest
     sourceWsCloudPlatform === destWsCloudPlatform
   );
 });
+
+// Returns a message explaining why the user can't edit the workspace, or undefined if they can
+export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }) => {
```

Comment:
It would be nice to add types to this function.

```suggestion
export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }: WorkspaceWrapper): string | undefined => {
```

Reply:
I'll give that try. As long as it doesn't substantially expand the scope, I'll add it to this one, otherwise in a follow-up.

Reply:
I took this out. I had to add `isLocked` to the `BaseWorkspaceInfo` interface. This would mean adding `isLocked: false` to several initialization spots, which increases the scope too much for my comfort. Happy to add this in a follow-up.

Reply:
I was going to make the same comment. It's a shame that `cond` can't produce the correct type using inference.

Reply:
I believe this is less related to `cond` and more related to `WorkspaceWrapper` and related types being built up gradually instead of up front, so fixtures have been created using an incomplete type.

As an alternative to using the full `WorkspaceWrapper` type here, we could follow @msilva-broad's suggestion of using only the slice of it that's required.

```suggestion
type EditWorkspaceErrorArg = {
  accessLevel: WorkspaceAccessLevel;
  workspace: {
    isLocked: boolean;
  };
};

export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }: EditWorkspaceErrorArg) => {
```

I think the trade-off is increased compatibility (no requirements for unnecessary fields) at the expense of clarity / types as documentation (it's no longer as clear that the function is intended to accept a `WorkspaceWrapper`).

Reply:
Another possible reason for preferring a partial workspace as the type... because of differing values for the `fields` parameter of [listWorkspaceDetails](https://rawls.dsde-prod.broadinstitute.org/#/workspaces/listWorkspaceDetails), parts of the app are often dealing with a partial workspace.

Reply:
But in any case, this is an improvement as-is. Figuring out that type can be left for another PR.